lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -0,0 +1,396 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file ParseXMLErrorHandler.cpp
+ */
+
+#include <utils/ParseXML.hpp>
+#include <utils/ParseXMLTags.hpp>
+#include <utils/ParseXMLString.hpp>
+
+bool initialize_tools (
+    xercesc::DOMConfiguration*& config,
+    xercesc::DOMDocument*& doc,
+    ParseXMLErrorHandler*& error_handler,
+    xercesc::DOMImplementation*& implementation,
+    xercesc::DOMLSOutput*& output,
+    xercesc::XercesDOMParser*& parser,
+    xercesc::DOMLSSerializer*& serializer,
+    xercesc::XMLFormatTarget*& target,
+    const std::string& xml_file
+)
+{
+    // Create Xerces implementation
+    implementation = xercesc::DOMImplementationRegistry::getDOMImplementation (xercesc::XMLString::transcode("Core"));
+    if (implementation == NULL)
+    {
+        // Could not initialize XML workspace
+        throw eprosima::qosprof::FileNotFound("Could not set XML workspace");
```

Comment:
This seem to be another exception different to FIleNotFound. Maybe we should define the Error exception for this case.